modules/loader/src/main/java/com/opengamma/strata/loader/csv/FullSwapTradeCsvLoader.java

```diff
@@ -147,27 +152,106 @@
    * @return the parsed trade
    */
   static SwapTrade parse(CsvRow row, TradeInfo info) {
-    List<RateCalculationSwapLeg> legs = new ArrayList<>();
     // parse any number of legs by looking for 'Leg n Pay Receive'
-    Optional<String> payReceive = Optional.of(getValue(row, "Leg 1 ", DIRECTION_FIELD));
+    // this finds the index for each leg, using null for fixed legs
+    List<FloatingRateIndex> indices = new ArrayList<>();
+    Set<DayCount> dayCounts = new LinkedHashSet<>();
+    boolean missingDayCount = false;
+    String legPrefix = "Leg 1 ";
+    Optional<String> payReceiveOpt = Optional.of(getValue(row, legPrefix, DIRECTION_FIELD));
     int i = 1;
-    while (payReceive.isPresent()) {
-      legs.add(parseLeg(row, "Leg " + i + " "));
+    while (payReceiveOpt.isPresent()) {
+      // parse this leg
+      FloatingRateIndex index = parseIndex(row, legPrefix);
+      if (index != null) {
+        // floating leg
+        indices.add(index);
+        dayCounts.add(index.getDefaultFixedLegDayCount());
+      } else {
+        // fixed leg
+        indices.add(null);
+        if (!findValue(row, legPrefix, DAY_COUNT_FIELD).isPresent()) {
+          missingDayCount = true;
```

Comment:
Do we even need the missingDayCount check? Feels like this is an unnecessary layer of complexity in what is already quite complex code.

What's the harm of passing a non-null default into the parseLegs method? Will just be ignored if the fixed leg has defined value.

Reply:
Case is where there is a fixed leg and two floating legs with different day counts. Must only trigger the "check day unique" logic when day count is actually missing.